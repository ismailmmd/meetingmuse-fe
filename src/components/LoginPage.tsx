import React from 'react';
import { LoginButton } from './AuthButtons';

interface LoginPageProps {
  onBack?: () => void;
}

export const LoginPage: React.FC<LoginPageProps> = ({ onBack }) => {
  return (
    <div className="min-h-screen bg-gray-50 flex flex-col justify-center items-center">
      <div className="max-w-md w-full space-y-8 p-8">
        {onBack && (
          <button
            onClick={onBack}
            className="flex items-center text-blue-600 hover:text-blue-800 transition-colors mb-4"
          >
            <svg className="w-4 h-4 mr-2" fill="none" stroke="currentColor" viewBox="0 0 24 24">
              <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M15 19l-7-7 7-7" />
            </svg>
            Back to Home
          </button>
        )}
        <div className="text-center">
          <h1 className="text-4xl font-bold text-gray-900 mb-2">MeetingMuse</h1>
        </div>
        
        <div className="bg-white p-8 rounded-lg shadow-md">
          <div className="text-center space-y-6">
            <div className="flex justify-center">
              <div className="w-16 h-16 bg-blue-100 rounded-full flex items-center justify-center">
                <svg className="w-8 h-8 text-blue-600" fill="currentColor" viewBox="0 0 20 20">
                  <path d="M2 5a2 2 0 012-2h7a2 2 0 012 2v4a2 2 0 01-2 2H9l-3 3v-3H4a2 2 0 01-2-2V5z" />
                  <path d="M15 7v2a4 4 0 01-4 4H9.828l-1.766 1.767c.28.149.599.233.938.233h2l3 3v-3h2a2 2 0 002-2V9a2 2 0 00-2-2h-1z" />
                </svg>
              </div>
            </div>
            
            <div>
              <h2 className="text-xl font-semibold text-gray-900 mb-2">
                Welcome to MeetingMuse
              </h2>
              <p className="text-gray-600 mb-6">
                Please sign in to start your conversation
              </p>
            </div>
            
            <LoginButton />
          </div>
        </div>
        
        <p className="text-center text-sm text-gray-500">
          Secure authentication powered by Google OAuth 2.0
        </p>
      </div>
    </div>
  );
};
