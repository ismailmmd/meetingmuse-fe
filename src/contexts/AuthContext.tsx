/* eslint-disable react-refresh/only-export-components */
import React, { createContext, useContext, useState, useEffect, useCallback, ReactNode } from 'react';
import { AuthService } from '../services/AuthService';

export interface User {
  id: string;
  email: string;
  name: string;
  picture?: string;
}

interface AuthSession {
  clientId: string;
  sessionId?: string;
  authenticated: boolean;
}

interface AuthContextType {
  user: User | null;
  isLoading: boolean;
  login: () => Promise<void>;
  logout: () => Promise<void>;
  isAuthenticated: boolean;
  session: AuthSession | null;
}

const AuthContext = createContext<AuthContextType | undefined>(undefined);

interface AuthProviderProps {
  children: ReactNode;
}

// Generate or get client ID for this browser session
const getClientId = (): string => {
  let clientId = localStorage.getItem('meetingmuse_client_id');
  if (!clientId) {
    clientId = `client-${Date.now()}-${Math.random().toString(36).substr(2, 9)}`;
    localStorage.setItem('meetingmuse_client_id', clientId);
  }
  return clientId;
};

export const AuthProvider: React.FC<AuthProviderProps> = ({ children }) => {
  const [user, setUser] = useState<User | null>(null);
  const [session, setSession] = useState<AuthSession | null>(null);
  const [isLoading, setIsLoading] = useState(true);

  const checkAuthStatus = async (clientId: string): Promise<void> => {
    try {
      const data = await AuthService.getAuthStatus(clientId);

      if (data.authenticated && data.session_id) {
        setSession({
          clientId,
          sessionId: data.session_id,
          authenticated: true,
        });

        // Create user data for authenticated session
        const tempUser: User = {
          id: data.session_id,
          email: 'user@authenticated.com',
          name: `Authenticated User (${clientId})`,
          picture: 'https://ui-avatars.com/api/?name=AU'
        };
        setUser(tempUser);
      } else {
        setSession({
          clientId,
          authenticated: false,
        });
        setUser(null);
      }
    } catch (error) {
      console.error('Error checking auth status:', error);
      setSession({
        clientId,
        authenticated: false,
      });
      setUser(null);
    }
  };

  const handleAuthCallback = useCallback(async (): Promise<void> => {
    const urlParams = new URLSearchParams(window.location.search);
    const authSuccess = urlParams.get('auth_success');
    const authError = urlParams.get('auth_error');

    if (authSuccess === 'true') {
      // Clear URL parameters
      window.history.replaceState({}, document.title, window.location.pathname);
      
      // Check auth status to get session info
      const clientId = getClientId();
      await checkAuthStatus(clientId);
    } else if (authError) {
      console.error('OAuth authentication error:', authError);
      // Clear URL parameters
      window.history.replaceState({}, document.title, window.location.pathname);
    }
  }, []);

  useEffect(() => {
    const initAuth = async () => {
      setIsLoading(true);
      const clientId = getClientId();
      
      // Handle OAuth callback if present
      await handleAuthCallback();
      
      // Check current auth status
      await checkAuthStatus(clientId);
      
      setIsLoading(false);
    };

    initAuth();
  }, [handleAuthCallback]);

  const login = async (): Promise<void> => {
    try {
      const clientId = getClientId();
      const data = await AuthService.startOAuthFlow(clientId);
      
      // Redirect to Google OAuth
      window.location.href = data.authorization_url;
    } catch (error) {
      console.error('Login failed:', error);
      throw error;
    }
  };

  const logout = async (): Promise<void> => {
    try {
      const clientId = getClientId();
      await AuthService.logout(clientId);
    } catch (error) {
      console.error('Logout request failed:', error);
    } finally {
      // Always clear local state
      setUser(null);
      setSession({
        clientId: getClientId(),
        authenticated: false,
      });
    }
  };

  const value: AuthContextType = {
    user,
    isLoading,
    login,
    logout,
    isAuthenticated: !!session?.authenticated && !!user,
    session,
  };

  return (
    <AuthContext.Provider value={value}>
      {children}
    </AuthContext.Provider>
  );
};

export const useAuth = (): AuthContextType => {
  const context = useContext(AuthContext);
  if (context === undefined) {
    throw new Error('useAuth must be used within an AuthProvider');
  }
  return context;
};
