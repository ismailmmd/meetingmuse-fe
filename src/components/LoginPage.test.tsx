import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen, waitFor } from '../test/test-utils';
import userEvent from '@testing-library/user-event';
import { LoginPage } from './LoginPage';
import { AuthProvider } from '../contexts/AuthContext';
import * as AuthService from '../services/AuthService';

// Mock the AuthService
vi.mock('../services/AuthService', () => ({
  AuthService: {
    startOAuthFlow: vi.fn(),
    getAuthStatus: vi.fn(),
    logout: vi.fn(),
  },
}));

// Mock localStorage
const localStorageMock = {
  getItem: vi.fn(),
  setItem: vi.fn(),
  removeItem: vi.fn(),
  clear: vi.fn(),
};
Object.defineProperty(window, 'localStorage', {
  value: localStorageMock,
});

// Wrapper component that provides AuthContext
const LoginPageWrapper = () => (
  <AuthProvider>
    <LoginPage />
  </AuthProvider>
);

describe('LoginPage', () => {
  beforeEach(() => {
    localStorageMock.getItem.mockClear();
    localStorageMock.setItem.mockClear();
    vi.mocked(AuthService.AuthService.getAuthStatus).mockResolvedValue({
      client_id: 'test-client-id',
      authenticated: false,
      message: 'Not authenticated',
    });
    vi.mocked(AuthService.AuthService.startOAuthFlow).mockResolvedValue({
      authorization_url:
        'https://accounts.google.com/oauth/authorize?test=true',
      state: 'test-state',
      client_id: 'test-client-id',
    });
  });

  describe('Snapshots', () => {
    it('should render the complete login page correctly', async () => {
      const { container } = render(<LoginPageWrapper />);

      // Wait for auth status check to complete
      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      expect(container.firstChild).toMatchSnapshot();
    });

    it('should render login page without back button', async () => {
      const { container } = render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      expect(container.firstChild).toMatchSnapshot('login-page-no-back-button');
    });

    it('should render login page with back link', async () => {
      const { container } = render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      expect(container.firstChild).toMatchSnapshot('login-page-with-back-link');
    });

    it('should match snapshot for main content card', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      const contentCard = screen
        .getByText('Welcome to MeetingMuse')
        .closest('.bg-white');
      expect(contentCard).toMatchSnapshot('login-content-card');
    });

    it('should match snapshot for back link', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Back to Home')).toBeInTheDocument();
      });

      const backLink = screen.getByRole('link', { name: 'Back to Home' });
      expect(backLink).toMatchSnapshot('back-link');
    });
  });

  describe('Content Rendering', () => {
    it('should display the main brand and welcome message', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('heading', { name: 'MeetingMuse', level: 1 })
        ).toBeInTheDocument();
        expect(
          screen.getByRole('heading', {
            name: 'Welcome to MeetingMuse',
            level: 2,
          })
        ).toBeInTheDocument();
        expect(
          screen.getByText(
            'Please sign in to start your conversation with our AI meeting assistant'
          )
        ).toBeInTheDocument();
      });
    });

    it('should display the chat icon', async () => {
      const { container } = render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      // Check for SVG icon presence
      const svgElements = container.querySelectorAll('svg');
      expect(svgElements.length).toBeGreaterThan(0);
    });

    it('should display the OAuth login button', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });
    });

    it('should display security message', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByText('Secure authentication powered by Google OAuth 2.0')
        ).toBeInTheDocument();
      });
    });
  });

  describe('Back Button Functionality', () => {
    it('should always render back link to home', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      const backLink = screen.getByRole('link', { name: 'Back to Home' });
      expect(backLink).toBeInTheDocument();
      expect(backLink).toHaveAttribute('href', '/');
    });

    it('should navigate to home when back link is clicked', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      const backLink = screen.getByRole('link', { name: 'Back to Home' });
      expect(backLink).toHaveAttribute('href', '/');
    });

    it('should display the correct back arrow icon', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      const backLink = screen.getByRole('link', { name: 'Back to Home' });
      expect(backLink).toContainHTML('stroke="currentColor"');
    });
  });

  describe('OAuth Login Functionality', () => {
    it('should display "Sign in with OAuth" button initially', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });

      const loginButton = screen.getByRole('button', {
        name: 'Sign in with OAuth',
      });
      expect(loginButton).toBeEnabled();
    });

    it('should show loading state when login is in progress', async () => {
      const user = userEvent.setup();

      // Mock a slower OAuth flow to test loading state
      vi.mocked(AuthService.AuthService.startOAuthFlow).mockImplementation(
        () =>
          new Promise((resolve) =>
            setTimeout(
              () =>
                resolve({
                  authorization_url:
                    'https://accounts.google.com/oauth/authorize',
                  state: 'test-state',
                  client_id: 'test-client',
                }),
              100
            )
          )
      );

      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });

      const loginButton = screen.getByRole('button', {
        name: 'Sign in with OAuth',
      });
      await user.click(loginButton);

      // Check loading state appears
      expect(screen.getByText('Signing in...')).toBeInTheDocument();
      expect(loginButton).toHaveClass('bg-gray-400', 'cursor-not-allowed');
    });

    it('should handle OAuth login button click', async () => {
      const user = userEvent.setup();

      // Mock successful OAuth flow
      const mockHref = vi.fn();
      Object.defineProperty(window, 'location', {
        value: { href: mockHref },
        writable: true,
      });

      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });

      const loginButton = screen.getByRole('button', {
        name: 'Sign in with OAuth',
      });
      await user.click(loginButton);

      await waitFor(() => {
        expect(AuthService.AuthService.startOAuthFlow).toHaveBeenCalled();
      });
    });

    it('should match snapshot for loading state', async () => {
      const user = userEvent.setup();

      // Mock delayed OAuth flow
      vi.mocked(AuthService.AuthService.startOAuthFlow).mockImplementation(
        () =>
          new Promise((resolve) =>
            setTimeout(
              () =>
                resolve({
                  authorization_url:
                    'https://accounts.google.com/oauth/authorize',
                  state: 'test-state',
                  client_id: 'test-client',
                }),
              1000
            )
          )
      );

      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });

      const loginButton = screen.getByRole('button', {
        name: 'Sign in with OAuth',
      });
      await user.click(loginButton);

      await waitFor(() => {
        expect(screen.getByText('Signing in...')).toBeInTheDocument();
      });

      const loadingButton = screen.getByRole('button', {
        name: 'Signing in...',
      });
      expect(loadingButton).toMatchSnapshot('oauth-button-loading');
    });
  });

  describe('Error Handling', () => {
    it('should handle OAuth flow errors gracefully', async () => {
      const user = userEvent.setup();

      // Mock OAuth flow failure
      vi.mocked(AuthService.AuthService.startOAuthFlow).mockRejectedValue(
        new Error('OAuth flow failed')
      );

      // Spy on console.error to verify error logging and suppress output
      const consoleSpy = vi
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });

      const loginButton = screen.getByRole('button', {
        name: 'Sign in with OAuth',
      });
      await user.click(loginButton);

      await waitFor(() => {
        expect(consoleSpy).toHaveBeenCalledWith(
          'Login error:',
          expect.any(Error)
        );
      });

      // Button should return to normal state after error
      expect(
        screen.getByRole('button', { name: 'Sign in with OAuth' })
      ).toBeInTheDocument();

      consoleSpy.mockRestore();
    });

    it('should reset button state after failed login', async () => {
      const user = userEvent.setup();

      // Mock OAuth flow failure
      vi.mocked(AuthService.AuthService.startOAuthFlow).mockRejectedValue(
        new Error('Network error')
      );

      // Suppress console errors for this test
      const consoleSpy = vi
        .spyOn(console, 'error')
        .mockImplementation(() => {});

      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });

      const loginButton = screen.getByRole('button', {
        name: 'Sign in with OAuth',
      });
      await user.click(loginButton);

      // Wait for error handling to complete
      await waitFor(() => {
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });

      // Button should be enabled again
      expect(loginButton).toBeEnabled();
      expect(loginButton).not.toHaveClass('cursor-not-allowed');

      consoleSpy.mockRestore();
    });
  });

  describe('Accessibility', () => {
    it('should have proper heading hierarchy', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        const h1 = screen.getByRole('heading', { level: 1 });
        const h2 = screen.getByRole('heading', { level: 2 });

        expect(h1).toBeInTheDocument();
        expect(h2).toBeInTheDocument();
      });
    });

    it('should have accessible button and link labels', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('link', { name: 'Back to Home' })
        ).toBeInTheDocument();
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });
    });

    it('should have proper color contrast in button states', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        const loginButton = screen.getByRole('button', {
          name: 'Sign in with OAuth',
        });
        expect(loginButton).toHaveClass('text-white');
        expect(loginButton).toHaveClass('bg-blue-600');
      });
    });

    it('should provide proper loading state accessibility', async () => {
      const user = userEvent.setup();

      // Mock delayed OAuth flow
      vi.mocked(AuthService.AuthService.startOAuthFlow).mockImplementation(
        () =>
          new Promise((resolve) =>
            setTimeout(
              () =>
                resolve({
                  authorization_url:
                    'https://accounts.google.com/oauth/authorize',
                  state: 'test-state',
                  client_id: 'test-client',
                }),
              100
            )
          )
      );

      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(
          screen.getByRole('button', { name: 'Sign in with OAuth' })
        ).toBeInTheDocument();
      });

      const loginButton = screen.getByRole('button', {
        name: 'Sign in with OAuth',
      });
      await user.click(loginButton);

      await waitFor(() => {
        const loadingButton = screen.getByRole('button', {
          name: 'Signing in...',
        });
        expect(loadingButton).toHaveAttribute('disabled');
      });
    });
  });

  describe('Visual Design Elements', () => {
    it('should include responsive design classes', async () => {
      const { container } = render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      expect(container.innerHTML).toContain('min-h-screen');
      expect(container.innerHTML).toContain('max-w-4xl');
      expect(container.innerHTML).toContain('px-4');
    });

    it('should include proper styling classes', async () => {
      const { container } = render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      expect(container.innerHTML).toContain('bg-gradient-to-br');
      expect(container.innerHTML).toContain('blue-gradient');
      expect(container.innerHTML).toContain('glass-effect');
      expect(container.innerHTML).toContain('rounded-xl');
    });

    it('should have proper button styling', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        const loginButton = screen.getByRole('button', {
          name: 'Sign in with OAuth',
        });
        expect(loginButton).toHaveClass(
          'bg-blue-600',
          'hover:bg-blue-700',
          'text-white'
        );
      });
    });
  });

  describe('Layout Structure', () => {
    it('should have proper page layout structure', async () => {
      render(<LoginPageWrapper />);

      await waitFor(() => {
        expect(screen.getByText('Welcome to MeetingMuse')).toBeInTheDocument();
      });

      // Check for header structure
      expect(screen.getByRole('banner')).toBeInTheDocument();

      // Check for main content structure
      expect(screen.getByRole('main')).toBeInTheDocument();

      // Check for content card with glass-effect
      const contentCard = screen
        .getByText('Welcome to MeetingMuse')
        .closest('.glass-effect');
      expect(contentCard).toBeInTheDocument();
    });
  });
});
