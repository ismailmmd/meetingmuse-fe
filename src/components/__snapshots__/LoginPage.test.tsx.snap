// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginPage > OAuth Login Functionality > should match snapshot for loading state > oauth-button-loading 1`] = `
<button
  class="
        px-6 py-3 rounded-lg font-medium text-white transition-colors
        bg-gray-400 cursor-not-allowed
      "
  disabled=""
>
  <div
    class="flex items-center"
  >
    <div
      class="w-4 h-4 border-2 border-white border-t-transparent rounded-full animate-spin mr-2"
    />
    Signing in...
  </div>
</button>
`;

exports[`LoginPage > Snapshots > should match snapshot for back link > back-link 1`] = `
<a
  class="flex items-center px-4 py-2 bg-white/20 hover:bg-white/30 backdrop-blur-sm rounded-lg text-white font-semibold transition-all duration-200 border border-white/30"
  data-discover="true"
  href="/"
>
  <svg
    class="w-4 h-4 mr-2"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
  >
    <path
      d="M15 19l-7-7 7-7"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
  Back to Home
</a>
`;

exports[`LoginPage > Snapshots > should match snapshot for main content card > login-content-card 1`] = `null`;

exports[`LoginPage > Snapshots > should render login page with back link > login-page-with-back-link 1`] = `
<div
  class="min-h-screen bg-gradient-to-br from-blue-50 via-white to-purple-50 flex flex-col"
>
  <header
    class="blue-gradient shadow-lg"
  >
    <div
      class="max-w-4xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between items-center py-6"
      >
        <div
          class="flex items-center space-x-4"
        >
          <div
            class="w-10 h-10 bg-white/20 backdrop-blur-sm rounded-lg flex items-center justify-center"
          >
            <svg
              class="w-5 h-5 text-white"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M2 5a2 2 0 012-2h7a2 2 0 012 2v4a2 2 0 01-2 2H9l-3 3v-3H4a2 2 0 01-2-2V5z"
              />
              <path
                d="M15 7v2a4 4 0 01-4 4H9.828l-1.766 1.767c.28.149.599.233.938.233h2l3 3v-3h2a2 2 0 002-2V9a2 2 0 00-2-2h-1z"
              />
            </svg>
          </div>
          <div>
            <h1
              class="text-2xl font-bold text-white tracking-tight"
            >
              MeetingMuse
            </h1>
            <p
              class="text-blue-100 text-sm font-medium"
            >
              AI-Powered Meeting Scheduling
            </p>
          </div>
        </div>
        <a
          class="flex items-center px-4 py-2 bg-white/20 hover:bg-white/30 backdrop-blur-sm rounded-lg text-white font-semibold transition-all duration-200 border border-white/30"
          data-discover="true"
          href="/"
        >
          <svg
            class="w-4 h-4 mr-2"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 19l-7-7 7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          Back to Home
        </a>
      </div>
    </div>
  </header>
  <main
    class="flex-1 max-w-4xl mx-auto px-4 sm:px-6 lg:px-8 py-12 w-full"
  >
    <div
      class="rounded-xl glass-effect p-8 mb-8 max-w-md mx-auto"
    >
      <div
        class="text-center space-y-6"
      >
        <div
          class="flex justify-center"
        >
          <div
            class="w-16 h-16 bg-blue-100 rounded-full flex items-center justify-center"
          >
            <svg
              class="w-8 h-8 text-blue-600"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M2 5a2 2 0 012-2h7a2 2 0 012 2v4a2 2 0 01-2 2H9l-3 3v-3H4a2 2 0 01-2-2V5z"
              />
              <path
                d="M15 7v2a4 4 0 01-4 4H9.828l-1.766 1.767c.28.149.599.233.938.233h2l3 3v-3h2a2 2 0 002-2V9a2 2 0 00-2-2h-1z"
              />
            </svg>
          </div>
        </div>
        <div>
          <h2
            class="text-2xl font-bold text-gray-900 mb-2"
          >
            Welcome to 
            MeetingMuse
          </h2>
          <p
            class="text-gray-600 mb-6"
          >
            Please sign in to start your conversation with our AI meeting assistant
          </p>
        </div>
        <button
          class="
        px-6 py-3 rounded-lg font-medium text-white transition-colors
        bg-blue-600 hover:bg-blue-700 active:bg-blue-800
      "
        >
          Sign in with OAuth
        </button>
        <p
          class="text-sm text-gray-500"
        >
          Secure authentication powered by Google OAuth 2.0
        </p>
      </div>
    </div>
  </main>
</div>
`;

exports[`LoginPage > Snapshots > should render login page without back button > login-page-no-back-button 1`] = `
<div
  class="min-h-screen bg-gradient-to-br from-blue-50 via-white to-purple-50 flex flex-col"
>
  <header
    class="blue-gradient shadow-lg"
  >
    <div
      class="max-w-4xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between items-center py-6"
      >
        <div
          class="flex items-center space-x-4"
        >
          <div
            class="w-10 h-10 bg-white/20 backdrop-blur-sm rounded-lg flex items-center justify-center"
          >
            <svg
              class="w-5 h-5 text-white"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M2 5a2 2 0 012-2h7a2 2 0 012 2v4a2 2 0 01-2 2H9l-3 3v-3H4a2 2 0 01-2-2V5z"
              />
              <path
                d="M15 7v2a4 4 0 01-4 4H9.828l-1.766 1.767c.28.149.599.233.938.233h2l3 3v-3h2a2 2 0 002-2V9a2 2 0 00-2-2h-1z"
              />
            </svg>
          </div>
          <div>
            <h1
              class="text-2xl font-bold text-white tracking-tight"
            >
              MeetingMuse
            </h1>
            <p
              class="text-blue-100 text-sm font-medium"
            >
              AI-Powered Meeting Scheduling
            </p>
          </div>
        </div>
        <a
          class="flex items-center px-4 py-2 bg-white/20 hover:bg-white/30 backdrop-blur-sm rounded-lg text-white font-semibold transition-all duration-200 border border-white/30"
          data-discover="true"
          href="/"
        >
          <svg
            class="w-4 h-4 mr-2"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 19l-7-7 7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          Back to Home
        </a>
      </div>
    </div>
  </header>
  <main
    class="flex-1 max-w-4xl mx-auto px-4 sm:px-6 lg:px-8 py-12 w-full"
  >
    <div
      class="rounded-xl glass-effect p-8 mb-8 max-w-md mx-auto"
    >
      <div
        class="text-center space-y-6"
      >
        <div
          class="flex justify-center"
        >
          <div
            class="w-16 h-16 bg-blue-100 rounded-full flex items-center justify-center"
          >
            <svg
              class="w-8 h-8 text-blue-600"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M2 5a2 2 0 012-2h7a2 2 0 012 2v4a2 2 0 01-2 2H9l-3 3v-3H4a2 2 0 01-2-2V5z"
              />
              <path
                d="M15 7v2a4 4 0 01-4 4H9.828l-1.766 1.767c.28.149.599.233.938.233h2l3 3v-3h2a2 2 0 002-2V9a2 2 0 00-2-2h-1z"
              />
            </svg>
          </div>
        </div>
        <div>
          <h2
            class="text-2xl font-bold text-gray-900 mb-2"
          >
            Welcome to 
            MeetingMuse
          </h2>
          <p
            class="text-gray-600 mb-6"
          >
            Please sign in to start your conversation with our AI meeting assistant
          </p>
        </div>
        <button
          class="
        px-6 py-3 rounded-lg font-medium text-white transition-colors
        bg-blue-600 hover:bg-blue-700 active:bg-blue-800
      "
        >
          Sign in with OAuth
        </button>
        <p
          class="text-sm text-gray-500"
        >
          Secure authentication powered by Google OAuth 2.0
        </p>
      </div>
    </div>
  </main>
</div>
`;

exports[`LoginPage > Snapshots > should render the complete login page correctly 1`] = `
<div
  class="min-h-screen bg-gradient-to-br from-blue-50 via-white to-purple-50 flex flex-col"
>
  <header
    class="blue-gradient shadow-lg"
  >
    <div
      class="max-w-4xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between items-center py-6"
      >
        <div
          class="flex items-center space-x-4"
        >
          <div
            class="w-10 h-10 bg-white/20 backdrop-blur-sm rounded-lg flex items-center justify-center"
          >
            <svg
              class="w-5 h-5 text-white"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M2 5a2 2 0 012-2h7a2 2 0 012 2v4a2 2 0 01-2 2H9l-3 3v-3H4a2 2 0 01-2-2V5z"
              />
              <path
                d="M15 7v2a4 4 0 01-4 4H9.828l-1.766 1.767c.28.149.599.233.938.233h2l3 3v-3h2a2 2 0 002-2V9a2 2 0 00-2-2h-1z"
              />
            </svg>
          </div>
          <div>
            <h1
              class="text-2xl font-bold text-white tracking-tight"
            >
              MeetingMuse
            </h1>
            <p
              class="text-blue-100 text-sm font-medium"
            >
              AI-Powered Meeting Scheduling
            </p>
          </div>
        </div>
        <a
          class="flex items-center px-4 py-2 bg-white/20 hover:bg-white/30 backdrop-blur-sm rounded-lg text-white font-semibold transition-all duration-200 border border-white/30"
          data-discover="true"
          href="/"
        >
          <svg
            class="w-4 h-4 mr-2"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 19l-7-7 7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          Back to Home
        </a>
      </div>
    </div>
  </header>
  <main
    class="flex-1 max-w-4xl mx-auto px-4 sm:px-6 lg:px-8 py-12 w-full"
  >
    <div
      class="rounded-xl glass-effect p-8 mb-8 max-w-md mx-auto"
    >
      <div
        class="text-center space-y-6"
      >
        <div
          class="flex justify-center"
        >
          <div
            class="w-16 h-16 bg-blue-100 rounded-full flex items-center justify-center"
          >
            <svg
              class="w-8 h-8 text-blue-600"
              fill="currentColor"
              viewBox="0 0 20 20"
            >
              <path
                d="M2 5a2 2 0 012-2h7a2 2 0 012 2v4a2 2 0 01-2 2H9l-3 3v-3H4a2 2 0 01-2-2V5z"
              />
              <path
                d="M15 7v2a4 4 0 01-4 4H9.828l-1.766 1.767c.28.149.599.233.938.233h2l3 3v-3h2a2 2 0 002-2V9a2 2 0 00-2-2h-1z"
              />
            </svg>
          </div>
        </div>
        <div>
          <h2
            class="text-2xl font-bold text-gray-900 mb-2"
          >
            Welcome to 
            MeetingMuse
          </h2>
          <p
            class="text-gray-600 mb-6"
          >
            Please sign in to start your conversation with our AI meeting assistant
          </p>
        </div>
        <button
          class="
        px-6 py-3 rounded-lg font-medium text-white transition-colors
        bg-blue-600 hover:bg-blue-700 active:bg-blue-800
      "
        >
          Sign in with OAuth
        </button>
        <p
          class="text-sm text-gray-500"
        >
          Secure authentication powered by Google OAuth 2.0
        </p>
      </div>
    </div>
  </main>
</div>
`;
